### ISSUE TITLE
Support ES6 Module Format?

### ISSUE THREAD

**Issue by MeirionHughes** [dcodeIO/protobuf.js#434]
The current exports are proving to be a little difficult to use within TypeScript + System.js module loader. Specifically, the way dcodeIO/\*  export a default function seem to make it problematic (both when typed out in TypeScript and then when you try to use it in System.JS). (I think this is the relevant issue: https://github.com/Microsoft/TypeScript/issues/5073)  

Would it be possible to have a separate es6-module format library?

**Comment by tmc** [comment]
@MeirionHughes why close this?

**Comment by MeirionHughes** [comment]
Well its something I suggested while struggling to get this working via JSPM, system.js and TypeScript, I fixed the import issue in the end though; (typings on dt are wrong). I'll reopen if its useful.

**Comment by backspaces** [comment]
I can see JSPM would be over-kill. Ditto webpack, browserify and the others. God, we spend way too much effort on workflow!

I've converted to using just system.js and, alas, babel-cli with transform-es2015-modules-systemjs. The latter *only* transforms import/export for system.js.  Hopefully <script type="module"... will land soon!

I use package.json for my build tool. The two sets of transformations, src/ and test/apps/ are built via:

    "build:as": "babel --plugins transform-es2015-modules-systemjs src -d lib",
    "build:test": "babel --plugins transform-es2015-modules-systemjs test/apps -d test/lib",
    "build": "npm run build:as && npm run build:test",

Does protobuf work with es6 modules via system.js? I've successfully used lots of non-es6 libraries using `import`.
